src/main.rs

```diff
@@ -12,23 +15,43 @@ enum Command {
     Login(login::ContestInfo),
 
     #[structopt(name = "submit", about = "Submits FILE to given PROBLEM.")]
-    Submit(submit::Submission),
+    Submit(client::Submission),
+}
+
+// TODO: Update all necessary code to support any judge system, not only Ejudge one
+fn from_config(config: ClientConfig) -> Result<Box<dyn SubmitClient>> {
+    Ok(Box::new(match config {
+        ClientConfig::Ejudge(config) => ejudge::EjudgeClient::from_config(config),
+        // TODO: add additional judge systems here
+    }?))
+}
+
+// TODO: Return client just built or something.
+async fn read_login(contest_info: &ContestInfo, into_path: &PathBuf) -> Result<()> {
```

Comment:
Why `read_login` instead of returning some useful data, writes all data to the config file and returns `()` instead of expected useful data? Rename this, please